fiftyone/utils/paddleocr_vl.py

```
@@ -0,0 +1,285 @@
+        img = np.repeat(img, 3, axis=2)
+    elif img.shape[2] == 4:
+        img = img[:, :, :3]
+    return PILImage.fromarray(img).convert("RGB")
+
+
+def _select_dtype(device) -> "torch.dtype":
+    """bfloat16 is a CUDA optimization; CPU inference wants float32."""
+    return torch.bfloat16 if "cuda" in str(device) else torch.float32
+
+
+def _upscale_for_spotting(pil):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `_upscale_for_spotting`

(ANN202)